Disher/View Controllers/RecipesViewController.m

```diff
@@ -294,34 +342,36 @@ - (CGRect)destinationFrameForSearchBar:(INSSearchBar *)searchBar {
 }
 
 - (void)searchBarDidTapReturn:(INSSearchBar *)searchBar {
-    //
+    //clear picker view
+    [self.cuisinesSet removeAllObjects];
+    [self.cuisines removeAllObjects];
+    self.cuisineField.text = @"";
+    [self.pickerView reloadAllComponents];
     if (self.searchSegmentedControl.selectedSegmentIndex == 0) {
         [self queryAPIs:searchBar.searchField.text withOption:0 completionHandler:^{
-            [self.tableView reloadData];
+            [self refreshData];
         }];
     }
     else {
         [self queryAPIs:searchBar.searchField.text withOption:1 completionHandler:^{
-            [self.tableView reloadData];
+            [self refreshData];
         }];
     }
 }
 
-//- (NSString *) parseUserInput:(NSString *) input {
-//    //get all the ingredients in an array
-//    NSArray *items = [input componentsSeparatedByString:@","];
-//    
-//    //remove duplicates in array
-//    return
-//}
+- (void) refreshData {
+    [self.tableView reloadData];
+    [self.cuisinesSet addObjectsFromArray:self.cuisines];
+    [self.cuisines setArray:[self.cuisinesSet allObjects]];
```

Comment:
What's the purpose of `cuisinesSet` vs `cuisines`?

Reply:
The cuisinesSet is there to get rid of duplicate entries. I am using the cuisines mutable array as the source for the picker view because I cannot dereference a set